### PR TITLE
fix: solve quantity false positives issue #2037

### PR DESCRIPTION
**Pull request by CharlesNepote**
**Description:**

Change regexp to avoid many false positives.

**Related issues and discussion:** #2037

### CODE REVIEW

**Comment by VaiTon** [comment]
@CharlesNepote could you provide some cases that the old regex matched and this one does not? Thank you! :-)

File: lib/ProductOpener/SiteQuality_off.pm [hunk 1]
```diff
@@ -738,13 +738,15 @@ sub check_ingredients($) {
...
+# Check quantity values. See https://en.wiki.openfoodfacts.org/Products_quantities
 sub check_quantity($) {
 
 	my $product_ref = shift;
 
 	# quantity contains "e" - might be an indicator that the user might have wanted to use "℮" \N{U+212E}
+	# example: 650 g e
 	if ((defined $product_ref->{quantity})
-		and ($product_ref->{quantity} =~ /(?:.*e$)|(?:[0-9]+\s*[kmc]?[gl]?\s*e)/i)
+		and ($product_ref->{quantity} =~ /(?:^e\s)|(?:\se[^a-z])|(?:\se$)/i)
```
Comment (stephanegigandet) [inline lib/ProductOpener/SiteQuality_off.pm]: The purpose of the more complex regexp was to flag things like this : "500 ge"

**Comment by stephanegigandet** [comment]
I think @VaiTon 's suggestion is a very good one: we should add test cases (both positives like "500 ge" and negatives like "1 litre") so that we don't break this later. There is already a test file : t/sitequality.t

**Comment by CharlesNepote** [comment]
Yes, you're both right. I'm not very comfortable with test cases but I know it's important, so I'll do it.

**Comment by teolemon** [comment]
rebased. @CharlesNepote can you write the test so we can merge

**Comment by teolemon** [comment]
Can we close @CharlesNepote ?

**Comment by benbenben2** [comment]
@CharlesNepote , @stephanegigandet , @VaiTon , @teolemon 

Tests have been added.
The regex (initially `/(?:.*e$)|(?:[0-9]+\s*[kmc]?[gl]?\s*e)/i`) have been updated. 
Because it seems to be covered already by the second part of the regex, the first part of the regex (`/(?:.*e$)|`) has been removed. 
Because this "e" letter is always (? could not find exception) preceded by "g" or "kg" or "l" or "ml", we can make `[gl]` as mandatory in the regex (by removing the question mark which makes it optional). This leads to: `/(?:[0-9]+\s*[kmc]?[gl]\s*e)/i`.

False positives such as "litre" or "verre" (ending by letter e) are not catch anymore with this regex.

**Review by stephanegigandet — APPROVED** [review]
Perfect, thank you!

**Pull request merged**